dll/win32/shell32/dialogs/dialogs.cpp

```diff
@@ -1692,3 +1684,23 @@ void WINAPI ExitWindowsDialog(HWND hWndOwner)
 
     FreeLibrary(msginaDll);
 }
+
+static BOOL IsShutdownAllowed(VOID)
+{
+    return SHTestTokenPrivilegeW(NULL, L"SeShutdownPrivilege");
+}
+
+/*************************************************************************
+ * ExitWindowsDialog                [SHELL32.60]
+ *
+ * @see http://www.jasinskionline.com/WindowsApi/ref/e/exitwindowsdialog.html
+ */
+void WINAPI ExitWindowsDialog(HWND hWndOwner)
+{
+    TRACE("(%p)\n", hWndOwner);
+
+    if (IsOS(OS_WELCOMELOGONUI) && !IsShutdownAllowed())
```

Comment:
Nice, but what about https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/shlwapi/ordinal.c;hb=d1e9fe13de06e9c6d17e80f2b3dd4804235b9650#l4090 ?
Also this is part of the "TODO" shown in the implement of this renamed ShutdownWindowsDialog function.

Note also this: https://www.geoffchappell.com/studies/windows/shell/shlwapi/api/isos/welcomelogonui.htm
but also that: https://learn.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-isos
> OS_WELCOMELOGONUI | 27 | Always returns FALSE.